client/lib/warn/index.js

```diff
@@ -5,9 +5,11 @@ var config = require( 'config' );
 
 function warn() {
 	if ( config( 'env' ) !== 'production' ) {
```

Comment:
>can config change over the life of the app? we could make warn a noop and then override it if we're not in prod

Sure, makes sense, and could shave a few cycles in production. What do you think about 726ccf7 ?